.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -88,6 +88,55 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           verbose: true
 
+  build-wasm:
+    needs: [clang_format, license_check, rust_format_check, markdown_lint, shellcheck]
+    runs-on: ubuntu-22.04
+    env:
+      SCCACHE_CACHE_SIZE: 600M
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - name: Setup toolchain
+        run: |
+          rustup component add clippy rustfmt
+          rustup target add wasm32-unknown-emscripten
+          git clone https://github.com/emscripten-core/emsdk
+          ./emsdk/emsdk install 2.0.14
+          ./emsdk/emsdk activate 2.0.14
+          source ./emsdk/emsdk_env.sh
+      - name: "Build Rust tools"
+        run: |
+          cargo install --no-default-features sccache
+          cargo install mdbook mdbook-linkcheck
```

Comment:
change to the similar to the matrix so that it uses caching (as this step takes 5.5 mins) and we can just reuse the existing Ubuntu cache for the tools

```
    - name: "Rust tools cache"
      uses: actions/cache@v4
      id: rust-tools-cache
      with:
        path: |
          ~/.cargo/bin/sccache
          ~/.cargo/bin/mdbook
          ~/.cargo/bin/mdbook-linkcheck
        key: ubuntu-22.04_sccache-0.7.6_mdbook-0.4.36_mdbook-linkcheck-0.7.7
    - name: "Build Rust tools"
      if: steps.rust-tools-cache.outputs.cache-hit != 'true'
      # Do not build with storage backends enabled, we only need local
      run: |
        cargo install --no-default-features sccache
        cargo install mdbook mdbook-linkcheck
```